### ISSUE TITLE
[Bug]: Random Validation Split Causes Temporal Leakage in price_forecaster.py

### ISSUE THREAD

**Issue by Tushar15769** [Eshajha19/agri#1444]
## 📌 Description
The LSTM training pipeline uses:
```
validation_split=0.1
```
for time-series data.

Random validation splitting allows future observations to appear in the validation set while related historical sequences remain in the training set.

Because temporal ordering is ignored, the model indirectly gains access to future information during training.

This results in overly optimistic validation metrics and compromises early stopping decisions.

---

## 🔁 Steps to Reproduce
Train the forecasting model on chronological data.
Use random validation splitting.
Inspect validation samples and training sequences.
Observe temporal overlap between training and validation periods.

---

## ❗ Expected Behavior
The application should:

preserve chronological ordering
use a time-based validation split
prevent future observations from influencing training
produce realistic validation metrics

---

## ⚠️ Actual Behavior
validation data is randomly selected
future observations can leak into training context
validation loss appears artificially low
early stopping decisions become unreliable
forecasting performance is overstated

---

## 📷 Screenshots / Logs
N/A

---

## 💻 Environment
- Device: (Desktop/Mobile)
- Browser: (Chrome/Firefox/etc.)
- OS: (Windows/Linux/Mac)

---

## 📢 Contribution Guidelines
- Comment **"assign me"** to work on this issue  
- Provide proper fix with explanation

**Comment by Tushar15769** [comment]
I would like to work upon this issue so please assign this issue to me @Eshajha19 .
Thanks!

**Comment by Eshajha19** [comment]
Assigned @Tushar15769
